src/serializer/ResidualFunctionInitializers.js

```diff
@@ -118,6 +119,12 @@ export class ResidualFunctionInitializers {
     return !!this.functionInitializerInfos.get(functionValue);
   }
 
+  factorifyInitializers(nameGenerator: NameGenerator) {
+    for (const initializer of this.initializers.values()) {
+      factorifyObjects(initializer.body, nameGenerator);
```

Comment:
Calling `factorifyObjects` multiple times is not the most optimal solution:
1. It may result in the creation of virtually identical factory function (duplicate code)
2. It's frequency-based heuristics will be distorted.

Doing it this way might be the easiest, but would it be possible instead to delay it until later and then do it all in one go?